pkg/forman/forman.go

```diff
@@ -0,0 +1,109 @@
+package forman
+
+import (
+	"time"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/utils"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/workers"
+
+	"go.uber.org/zap"
+)
+
+type Request struct {
+	Name string `json:"name"`
+}
+
+type Result struct {
+	Requeue bool
+	After   time.Duration
+}
+
+type provider struct {
+	working    bool
+	next       *time.Time
+	alarmClock *time.Timer
+}
+
+func GoForman(
+	logger *zap.Logger,
+	dbInstance *db.Postgres,
+	workerCount int,
+	reconcile func(*zap.Logger, *db.Postgres, Request) Result,
```

Comment:
It is a bad idea to couple the Forman with the DB instance or have the reconcile function have db-related things in the callback signature. 

I might see a case for Logger here, but even that is a stretch

Reply:
db is a concurrent safe operation instance, do you want it to be removed for now?